### PR TITLE
HIPANGULAR-62 adapt API calls to JSON

### PR DESCRIPTION
**Pull request by maltehol**

### CODE REVIEW

**Comment by sstamm** [comment]
Related WebAPI changes:
https://github.com/HiP-App/HiP-CmsWebApi/pull/47
Should be merged at the same time.

File: app/topics/shared/topic.service.ts [hunk 1]
```diff
@@ -221,8 +220,12 @@ export class TopicService {
    * @returns {Promise<Topic>} a Promise for a Topic object
    */
   public updateTopic(topic: Topic) {
-    let data = topic.formData();
-    return this.cmsApiService.putUrl('/api/Topics/' + topic.id, data, {})
+    let topicJson = JSON.stringify(topic, function(key, value) {
+      if(key === 'students' || key === 'supervisor' || key === 'reviewer') {
```
Comment (HIPCMSOrgaUser) [inline app/topics/shared/topic.service.ts]: __[tslint]__
* missing whitespace

File: app/shared/taginput/user-tag-input.component.ts [hunk 2]
```diff
@@ -31,7 +39,21 @@ export class UserTagInputComponent implements OnInit {
   constructor(private userService: UserService) {}
 
   ngOnInit() {
+    console.log('onInit');
```
Comment (HIPCMSOrgaUser) [inline app/shared/taginput/user-tag-input.component.ts]: __[tslint]__
* function invocation disallowed: console.log

File: app/shared/taginput/user-tag-input.component.ts [hunk 3]
```diff
@@ -31,7 +39,21 @@ export class UserTagInputComponent implements OnInit {
...
     this.tagPlaceholder = ' +' + this.role;
+    console.log(this.users);
```
Comment (HIPCMSOrgaUser) [inline app/shared/taginput/user-tag-input.component.ts]: __[tslint]__
* function invocation disallowed: console.log

File: app/shared/taginput/user-tag-input.component.ts [hunk 4]
```diff
@@ -31,7 +39,21 @@ export class UserTagInputComponent implements OnInit {
...
+
+  ngAfterViewChecked() {
+    for (let user of this.users) {
+      if (user.picture === undefined) {
+        this.userService.getPicture(user.id + '')
+          .then((response: any) => {
+            user.picture = response.json().base64
+          })
+      }
+      console.log(user.picture);
```
Comment (HIPCMSOrgaUser) [inline app/shared/taginput/user-tag-input.component.ts]: __[tslint]__
* function invocation disallowed: console.log

File: app/shared/taginput/user-tag-input.component.ts [hunk 5]
```diff
@@ -31,7 +39,21 @@ export class UserTagInputComponent implements OnInit {
...
+    console.log(this.users);
+  }
+
+  ngAfterViewChecked() {
+    for (let user of this.users) {
+      if (user.picture === undefined) {
+        this.userService.getPicture(user.id + '')
+          .then((response: any) => {
+            user.picture = response.json().base64
+          })
```
Comment (HIPCMSOrgaUser) [inline app/shared/taginput/user-tag-input.component.ts]: __[tslint]__
* missing semicolon

File: app/shared/taginput/user-tag-input.component.ts [hunk 6]
```diff
@@ -31,7 +39,21 @@ export class UserTagInputComponent implements OnInit {
...
     this.tagPlaceholder = ' +' + this.role;
+    console.log(this.users);
+  }
+
+  ngAfterViewChecked() {
+    for (let user of this.users) {
+      if (user.picture === undefined) {
+        this.userService.getPicture(user.id + '')
+          .then((response: any) => {
+            user.picture = response.json().base64
```
Comment (HIPCMSOrgaUser) [inline app/shared/taginput/user-tag-input.component.ts]: __[tslint]__
* missing semicolon

File: app/shared/taginput/user-tag-input.component.ts [hunk 7]
```diff
@@ -34,6 +42,17 @@ export class UserTagInputComponent implements OnInit {
     this.tagPlaceholder = ' +' + this.role;
   }
 
+  ngAfterViewChecked() {
+    for (let user of this.users) {
+      if (user.picture === undefined) {
+        this.userService.getPicture(user.id + '')
+          .then((response: any) => {
+            user.picture = response.json().base64;
+          })
```
Comment (HIPCMSOrgaUser) [inline app/shared/taginput/user-tag-input.component.ts]: __[tslint]__
* missing semicolon

**Pull request merged**